src/strings_type.h

```diff
@@ -71,6 +72,52 @@ static constexpr StringID SPECSTR_SILLY_NAME = 0x70E5; ///< Special string for s
 static constexpr StringID SPECSTR_ANDCO_NAME = 0x70E6; ///< Special string for Surname & Co company names.
 static constexpr StringID SPECSTR_PRESIDENT_NAME = 0x70E7; ///< Special string for the president's name.
 
-using StringParameterData = std::variant<uint64_t, std::string>;
+using StringParameterData = std::variant<std::monostate, uint64_t, std::string>;
+
+/** The data required to format and validate a single parameter of a string. */
+struct StringParameter {
+	StringParameterData data; ///< The data of the parameter.
+	char32_t type; ///< The #StringControlCode to interpret this data with when it's the first parameter, otherwise '\0'.
+
+	StringParameter() = default;
+	inline StringParameter(StringParameterData &&data) : data(std::move(data)), type(0) {}
+
+	inline StringParameter(const std::monostate &data) : data(data), type(0) {}
+	inline StringParameter(uint64_t data) : data(data), type(0) {}
+
+	inline StringParameter(const char *data) : data(std::string{data}), type(0) {}
+	inline StringParameter(std::string &&data) : data(std::move(data)), type(0) {}
+	inline StringParameter(const std::string &data) : data(data), type(0) {}
+
+	template <typename T, std::enable_if_t<std::is_base_of<StrongTypedefBase, T>::value, int> = 0>
+	inline StringParameter(const T &data) : data(static_cast<uint64_t>(data.base())), type(0) {}
```

Comment:
```suggestion
	inline StringParameter(const ConvertibleThroughBase auto &data) : data(static_cast<uint64_t>(data.base())), type(0) {}
```
And potentially some other places.